packages/network/samba/package.mk

```diff
@@ -24,7 +24,7 @@ PKG_LICENSE="GPLv3+"
 PKG_SITE="https://www.samba.org"
 PKG_URL="https://download.samba.org/pub/samba/stable/$PKG_NAME-$PKG_VERSION.tar.gz"
 PKG_DEPENDS_TARGET="toolchain heimdal:host e2fsprogs Python2 zlib readline popt libaio connman"
-PKG_NEED_UNPACK="$(get_pkg_directory heimdal) $(get_pkg_directory e2fsprogs)"
+PKG_NEED_UNPACK="$DISTRO_DIR/$DISTRO/config/smb.conf $(get_pkg_directory heimdal) $(get_pkg_directory e2fsprogs)"
```

Comment:
Strictly speaking, if we're observing changes of the distro config file in order to trigger a package clean, then why not also the equivalent [`PROJECT` config files](https://github.com/Kwiboo/LibreELEC.tv/blob/rebuild-package/packages/network/samba/package.mk#L146-L151) that could exist? Either we add all three file locations, or none at all. I'd probably go with the latter - is this strictly necessary?

Reply:
Unrelated to this PR but I've noticed that any non-`$PKG_DIR/config/smb.conf` will not install `samba.conf.sample`, which isn't really a problem but could be confusing. Not sure why we did that, assume it's an oversight?